File: src/main/java/com/example/SEB/dto/SignUpRequest.java
package com.example.SEB.dto;

import java.util.Date;

import com.example.SEB.enums.UserRole;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.Data;

@Data
public class SignUpRequest {
    
    private String username;
    private String email;
    private String password;
    private String phoneNumber;
    private Date createDate;
    @Enumerated(EnumType.STRING)
    private UserRole role;
}

File: seb_frontend/src/app/modules/admin/components/dashboard/SignUpRequest.ts
import { UserRole } from "./user-role";

  export class SignUpRequest {
    username: string;
    email: string;
    password: string;
    phoneNumber: string;
    role: UserRole;
    isActive: boolean;

    constructor(
      username: string,
      email: string,
      password: string,
      phoneNumber: string,
      role: UserRole,
      isActive: boolean = false,
    ) {
      this.username = username;
      this.email = email;
      this.password = password;
      this.phoneNumber = phoneNumber;
      this.role = role;
      this.isActive = isActive;
    }
  }

export { UserRole };
